app/src/main/java/net/demilich/metastone/gui/IconFactory.java

```diff
@@ -48,80 +48,101 @@ public static Image getDialogIcon(DialogType dialogType) {
 
 	public static String getHeroIconUrl(HeroClass heroClass) {
 		String iconPath = RESOURCE_PATH + "/img/heroes/";
-		switch (heroClass) {
-		case DRUID:
-			iconPath += "malfurion";
-			break;
-		case HUNTER:
-			iconPath += "rexxar";
-			break;
-		case MAGE:
-			iconPath += "jaina";
-			break;
-		case PALADIN:
-			iconPath += "uther";
-			break;
-		case PRIEST:
-			iconPath += "anduin";
-			break;
-		case ROGUE:
-			iconPath += "valeera";
-			break;
-		case SHAMAN:
-			iconPath += "thrall";
-			break;
-		case WARLOCK:
-			iconPath += "guldan";
-			break;
-		case WARRIOR:
-			iconPath += "garrosh";
-			break;
-		default:
-		case ANY:
-			iconPath += "unknown";
-			break;
-
+		if (((HeroCard)hero.getSourceCard()).getAsset().equals("<Def>")){
+			switch (hero.getHeroClass()) {
+			case DRUID:
+				iconPath += "malfurion";
+				break;
+			case HUNTER:
+				iconPath += "rexxar";
+				break;
+			case MAGE:
+				iconPath += "jaina";
+				break;
+			case PALADIN:
+				iconPath += "uther";
+				break;
+			case PRIEST:
+				iconPath += "anduin";
+				break;
+			case ROGUE:
+				iconPath += "valeera";
+				break;
+			case SHAMAN:
+				iconPath += "thrall";
+				break;
+			case WARLOCK:
+				iconPath += "guldan";
+				break;
+			case WARRIOR:
+				iconPath += "garrosh";
+				break;
+			default:
+			case ANY:
+				iconPath += "unknown";
+				break;
+	
+			}
+			return iconPath + ".png";
+		} else {
+			String assetsFolder = UserHomeMetastone.getPath().replace("\\\\", "/") + "/" + "assets";
+			String pth = assetsFolder + "/" + ((HeroCard)hero.getSourceCard()).getAsset() + ".png";
+			try {
+				return new File(pth).toURI().toURL().toString();
+			} catch (MalformedURLException e) {
+				return pth;
+			}
 		}
-		return iconPath + ".png";
 	}
 
 	public static String getHeroPowerIconUrl(HeroPower heroPower) {
-		String iconPath = RESOURCE_PATH + "/img/powers/";
-		switch (heroPower.getHeroClass()) {
-		case DRUID:
-			iconPath += "shapeshift";
-			break;
-		case HUNTER:
-			iconPath += "steady_shot";
-			break;
-		case MAGE:
-			iconPath += "fireblast";
-			break;
-		case PALADIN:
-			iconPath += "reinforce";
-			break;
-		case PRIEST:
-			iconPath += "lesser_heal";
-			break;
-		case ROGUE:
-			iconPath += "dagger_mastery";
-			break;
-		case SHAMAN:
-			iconPath += "totemic_call";
-			break;
-		case WARLOCK:
-			iconPath += "life_tap";
-			break;
-		case WARRIOR:
-			iconPath += "armor_up";
-			break;
-		default:
-			iconPath += "unknown";
-			break;
-
+		if (heroPower.getAsset().equals("<Def>")){
```

Comment:
I'm assuming you declared "<Def>" to mean default value?  Could we not use `null` to mean default value?  In which case this becomes a null check rather than a string comparison.  A tad bit more efficient :)  If I misunderstood the intention of "<Def>" or I have overlooked something in your PR as to why it's needed, please explain.